Lacework/lacework_module/lacework_connector.py

```diff
@@ -121,55 +138,58 @@ def get_next_events(self, cursor: str | None) -> Any | None:
 
-            items = batch.get("data", [])
-            if len(items) > 0:
-                most_recent_timestamp_seen = self._get_most_recent_timestamp_from_items(items)
-                events_lag = int(time.time() - most_recent_timestamp_seen)
-                EVENTS_LAG.labels(intake_key=self.configuration.intake_key).set(events_lag)
-                INCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(items))
+            last_date = self._get_most_recent_timestamp_from_items(items)
+            events_lag = int(time.time() - last_date)
+            EVENTS_LAG.labels(intake_key=self.configuration.intake_key).set(events_lag)
 
-            for message in items:
-                messages.append(orjson.dumps(message).decode("utf-8"))
-
-            if len(messages) > self.pagination_limit:
-                self.log(message=f"Sending a batch of {len(messages)} messages", level="info")
-                OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(messages))
-                self.push_events_to_intakes(events=messages)
-                messages = []
-
-        self.cursor = cursor
-
-        if messages:
-            self.log(message=f"Sending a batch of {len(messages)} messages", level="info")
+            messages = [orjson.dumps(message).decode("utf-8") for message in items]
+            self.log(message=f"Sending the first batch of {len(messages)} messages", level="info")
             OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(messages))
             self.push_events_to_intakes(events=messages)
+
+            grouped_data = []
+            while next_page_url:
+                response_next_page = self.get_next_events(next_page_url)
+                if response_next_page:
+                    next_page_url = (
+                        response_next_page.get("content", {}).get("paging", {}).get("urls", {}).get("nextPage")
+                    )
+                    next_page_items = response_next_page.get("items", [])
+                    next_page_items = [orjson.dumps(message).decode("utf-8") for message in next_page_items]
+                    grouped_data.extend(next_page_items)
+
+                    if len(grouped_data) > self.pagination_limit:
+                        self.log(message=f"Sending a batch of {len(grouped_data)} messages", level="info")
+                        OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(grouped_data))
+                        self.push_events_to_intakes(events=grouped_data)
+                        grouped_data = []
+
+            if grouped_data != []:
+                self.log(message=f"Sending a batch of {len(grouped_data)} messages", level="info")
+                OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(grouped_data))
+                self.push_events_to_intakes(events=grouped_data)
+
+            self.most_recent_date_seen = now
```

Comment:
`most_recent_date_seen` must host the most recent date seen in the events. I suggest changing the way to value this property
```suggestion
```